apm-sniffer/apm-sdk-plugin/dbcp-2.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/dbcp/v2/define/BasicDataSourceInstrumentation.java

```diff
@@ -72,4 +90,9 @@ public boolean isOverrideArgs() {
     public StaticMethodsInterceptPoint[] getStaticMethodsInterceptPoints() {
         return new StaticMethodsInterceptPoint[0];
     }
+
+    @Override
+    public boolean isBootstrapInstrumentation() {
+        return true;
```

Comment:
Why need bootstrap instrumentation? Is this a mischange?

Reply:
Without rewriting, enhancement does not work

Reply:
It doesn't make sense. Bootstrap instrumentation is for rt.jar, such as javax.*

Reply:
Check all existing plugins, there are very few activating this setting. And for DBCP, druid and hikaricp existing plugins, all of them don't require this. Please recheck.

Reply:
I can see you delete one, but keep the other two?

Reply:
> I can see you delete one, but keep the other two?

Yes, multiple verification deletions cannot be enhanced

Reply:
I am afraid you have to check. This is not right in theory. 
Let's take DBCP plugin as an example, the interceptor and instrumentation existed for months, and are verified by plugin e2e tests in every commit. Why does your instrumentation not work? 
I have to ask you to example, from my agent core experience, this is not the right way to write a plugin.

Reply:
> I am afraid you have to check. This is not right in theory. Let's take DBCP plugin as an example, the interceptor and instrumentation existed for months, and are verified by plugin e2e tests in every commit. Why does your instrumentation not work? I have to ask you to example, from my agent core experience, this is not the right way to write a plugin.

I debug, screenshot

Reply:
Again, compare to exiting codes and plugins. And begin to write the official plugin e2e tests. The existing codes there have verified, this option is not required to return `true`. I have to insist on this way.

Reply:
@sunOnly Why does DBCP plugin still keep this?

Reply:
> @sunOnly Why does DBCP plugin still keep this?

The local test does not take effect. See if there is a better entry point

Reply:
This class has been instrumented by the existing plugin, you are just adding another method. This should work. If not, I am guessing your test case is special? Did you debug for the existing interceptor?